includes/customers/customer-actions.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Customer Actions.
+ *
+ * This file contains all of the first class functions for interacting with a
+ * customer or its related meta data.
+ *
+ * @package     EDD
+ * @subpackage  Actions
+ * @copyright   Copyright (c) 2020, Sandhills Development, LLC
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       3.0
+ */
+
+/**
+ * When a new primary address is added to the database, any other primary addresses should be demoted.
+ *
+ * @param string $old_value
+ * @param string $new_value
+ * @param int $item_id
+ * @return void
+ */
+function edd_demote_customer_primary_addresses( $old_value, $new_value, $item_id ) {
+	if ( ! $new_value ) {
+		return;
+	}
+	$customer_addresses = new EDD\Database\Queries\Customer_Address();
+	$address            = $customer_addresses->get_item( $item_id );
+	if ( ! $address->is_primary ) {
+		return;
+	}
```

Comment:
I don't think we should need this query for fetching the whole object. `$new_value` should already be the same as `$address->is_primary` (same value). So querying it again shouldn't be needed.

Reply:
I agree and thought about removing it, but the object is required to get the `customer_id` parameter for the following query, so I left it in. I do agree that the `$address->is_primary` check is redundant.